tasklist/webapp/src/test/java/io/camunda/tasklist/webapp/es/backup/os/BackupManagerOpenSearchTest.java

```diff
@@ -145,4 +164,82 @@ public void shouldForwardPatternToOS() throws IOException {
+    doReturn(List.of(snapshotInfo)).when(backupManagerOpenSearch).findSnapshots(backupId);
+
+    final boolean finished =
+        backupManagerOpenSearch.isSnapshotFinishedWithinTimeout(snapshotInfo.snapshot());
+
+    assertFalse(finished);
+    verify(backupManagerOpenSearch, atLeast(5)).findSnapshots(backupId);
+  }
+
+  @Test
+  public void shouldWaitForSnapshotTillCompleted() throws IOException {
+    final int timeout = 0;
+    when(tasklistProperties.getBackup().getSnapshotTimeout()).thenReturn(timeout);
+    final int numberOfSnapshots = 6;
+    final var snapshotInfos = new ArrayList<SnapshotInfo>();
+
+    final List<Metadata> metadata = new ArrayList<>();
+    for (int i = 0; i < numberOfSnapshots; i++) {
+      metadata.add(
+          new Metadata()
+              .setBackupId(backupId)
+              .setPartCount(numberOfSnapshots)
+              .setPartNo(i)
+              .setVersion("8.3.0"));
+    }
+
+    for (int i = 0; i < numberOfSnapshots; i++) {
+      final SnapshotInfo snapshotInfo = mock(SnapshotInfo.class, RETURNS_DEEP_STUBS);
+      when(snapshotInfo.state())
+          .thenReturn(SnapshotState.STARTED.name())
+          .thenReturn(SnapshotState.STARTED.name())
+          .thenReturn(SnapshotState.SUCCESS.name());
+      when(snapshotInfo.snapshot()).thenReturn(metadata.get(i).buildSnapshotName());
+      snapshotInfos.add(snapshotInfo);
+    }
+
+    final CompletableFuture<CreateSnapshotResponse> failedResponse =
+        CompletableFuture.failedFuture(new SocketTimeoutException("Request timed out."));
+    when(openSearchAsyncClient.snapshot().create(any(CreateSnapshotRequest.class)))
+        .thenReturn(failedResponse);
+    when(openSearchClient._transport().performRequest(any(), any(), any()))
+        .thenReturn(new Builder().snapshots(snapshotInfos).build());
+
+    for (int i = 0; i < numberOfSnapshots; i++) {
+      final Metadata snapshotMetadata = metadata.get(i);
+      final String snapshotName = snapshotMetadata.buildSnapshotName();
+      backupManagerOpenSearch.executeSnapshotting(
+          CreateSnapshotRequest.of(
+              csr ->
+                  csr.repository(repositoryName)
+                      .snapshot(snapshotName)
+                      .indices(List.of("index-1", "index-2"))
+                      .ignoreUnavailable(false)
+                      .includeGlobalState(true)
+                      .metadata(
+                          Map.of(
+                              "backupId", JsonData.of(snapshotMetadata.getBackupId()),
+                              "version", JsonData.of(snapshotMetadata.getVersion()),
+                              "partNo", JsonData.of(snapshotMetadata.getPartNo()),
+                              "partCount", JsonData.of(snapshotMetadata.getPartCount())))
+                      .featureStates("none")
+                      .waitForCompletion(true)));
+    }
+
+    verify(backupManagerOpenSearch, atLeast(12)).findSnapshots(backupId);
```

Comment:
```suggestion
    verify(backupManagerOpenSearch, atLeast(2 * numberOfSnapshots)).findSnapshots(backupId);
```

Reply:
:thought_balloon:  Actually, I'm not sure why `2*numberOfSnapshot` and not `3*numberOfSnapshots` ?